### ISSUE TITLE
Can't use cross-filtering on a column from a table whose name has been overidden

### ISSUE THREAD

**Issue by martin-raymond** [apache/superset#24836]
cross_filter_works.mpa shows that you can cross filter the dashboard by choosing a line from the column 'filArianeContenu' from the table 'Liste de tous les éléments consultés (site courant)'

But, when you overide the name of a column to have a more user-friendly name 'A specific name user friendly', cross_filter_does_not_work.mpa shows that cross-filtering does not work properly.

### Environment

- superset version: `3.0.0rc2`

https://github.com/apache/superset/assets/10060967/746bdbe5-19de-49d4-850b-25081dfe86b4
https://github.com/apache/superset/assets/10060967/8e19a19f-ac16-4f9c-a48e-5ab60086abbf

**Comment by michael-s-molina** [comment]
@rusackas Who's the best person to investigate this issue?

**Comment by martin-raymond** [comment]
@rusackas did you find that person ?

**Comment by rusackas** [comment]
Nope! Not sure who that would be, but it seems the videos don't work here. If anyone can update this thread (or file a new issue) with a reproducible use case (preferably using example data/dashboards) that would be immensely helpful.

**Comment by rusackas** [comment]
Spoke too soon. The videos just took _forever_ to load. I'm not sure if this is the priority list for anyone at the moment, unless @yousoph knows otherwise. If anyone wants to investigate and open a PR, it would be hugely appreciated.

**Comment by fmannhardt** [comment]
I was encountering the same issue right now and it is very annoying when using tables. Just a thought on fixing this, maybe instead of renaming the columns (in SQL) to have them show user friendly in the UI, the table widget could be extended to allow for a UI only change in the "Customize column".

**Comment by rusackas** [comment]
This is still the case, and I was just getting annoyed by it myself. There may be a few options:
• Allow (by default or some control) cross filters to emit the underlying column name rather than the title
• Add UI to override the column name in the Table display so that you can adjust table column names (my usual reason for renaming the columns, personally)

Maybe @kasiazjc has thought about this?

**Comment by cheepon** [comment]
Any progress? It's been a long long time.

**Comment by rusackas** [comment]
To summarize the two approaches discussed for fixing this:
1. **Cross-filter emits underlying column name**: When a cross-filter is applied from a table chart, emit the raw column name (not the display label) so downstream filters can match correctly. This may require tracking a mapping from display label to column name in the chart's cross-filter output.
2. **UI-only column renaming**: Add a "display name" field to table column customization that changes what's shown in the table without modifying the SQL alias, so the technical column name is preserved end-to-end.

Either approach would be a valuable contribution. If you're interested in working on this, please comment here with your proposed approach before starting. @kasiazjc — any thoughts on direction?

**Comment by rusackas** [comment]
This shipped in #38858. Also, we no longer support 3.0, just to say that official bit. Table cross-filters now resolve a renamed column's label back to the real column name before emitting the filter. Please feel free to comment if I'm off-base here, or reopen with a fresh repro on current master if it's still misbehaving for you.
